protobuf-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/ProtobufSchemaConverter.java

```diff
@@ -69,7 +94,16 @@ public byte[] fromConnectData(String topic, Schema schema, Object value) {
     @Override
     public SchemaAndValue toConnectData(String topic, byte[] bytes) {
         final Message message = (Message) deserializer.deserialize(topic, bytes);
-        final Schema schema = protobufSchemaToConnectSchemaConverter.toConnectSchema(message);
+
+        final Descriptors.Descriptor descriptor = message.getDescriptorForType();
+        final Schema cachedConnectSchema = toConnectSchemaCache.get(descriptor);
+        final Schema schema;
```

Comment:
Why do we need to have both `cachedConnectSchema` and `schema`? One should be sufficient.

Reply:
Good point, fixed